### PR TITLE
[pre-commit] Add pre-commit and makes it happy.

### PR DESCRIPTION
**Pull request by olivier-stasse**
This PR aims at:
 * putting pre-commit in dynamic_graph_bridge such that next PR can be easily checked by maintainers and provides a consistent code format.
 * add a ros_publish test (this is a partial addition) it has yet to be completed.
 * start to add explanation on how to run tests.

The PR is incomplete, but should be included in humble before moving on further developments.

### CODE REVIEW

**Comment by nim65s** [comment]
I thirnk you should take the one from devel, or document why you don't

**Comment by olivier-stasse** [comment]
> I thirnk you should take the one from devel, or document why you don't

devel should be for rolling.
This one targets humble (LTS).
But I am planning to go for rolling with devel.

**Comment by olivier-stasse** [comment]
I have some strange behaviors with pre-commits. It works on the [github action](https://results.pre-commit.ci/run/github/1114754/1680531201.UHWHY2fmS_G-tPjz7VSeEA) but not on our [build farm](https://gitlab.laas.fr/stack-of-tasks/dynamic_graph_bridge/-/jobs/197572). @nim65s what is your recommendation ?

**Comment by nim65s** [comment]
> devel should be for rolling. This one targets humble (LTS).

Yes, but the .pre-commit-config.yaml file can be the the same. And should, IMO.

**Comment by olivier-stasse** [comment]
> > devel should be for rolling. This one targets humble (LTS).
> 
> Yes, but the .pre-commit-config.yaml file can be the the same. And should, IMO.

Sorry I completely misunderstood your first reply.
Yes it definitely should.

**Comment by olivier-stasse** [comment]
> > > devel should be for rolling. This one targets humble (LTS).
> > 
> > 
> > Yes, but the .pre-commit-config.yaml file can be the the same. And should, IMO.
> 
> Sorry I completely misunderstood your first reply. Yes it definitely should.

Even after using the same file, and adding the ignore flags for E501 and W503 for flake8, there is still a difference between the github action and our gitlab-ci.

**Comment by olivier-stasse** [comment]
@nim65s @florent-lamiraux I think that now this PR is clean in terms of pre-commit.

It adds some information in the badges and the README.md but in order to move forward with the tests. I will merge it today to advance unless you have a different point of view.

I added the .git-blame-ignore-revs file to avoid having the pre-commit style commits to pollute the history.

The test on ros_publish is still incomplete, but this PR is needed before moving on.

File: CMakeLists.txt [hunk 1]
```diff
@@ -31,9 +31,9 @@ include(cmake/ros.cmake)
 compute_project_args(PROJECT_ARGS LANGUAGES CXX)
 project(${PROJECT_NAME} ${PROJECT_ARGS})
 
-if(NOT CMAKE_CXX_STANDARD)
-  set(CMAKE_CXX_STANDARD 14)
-endif()
+# if(NOT CMAKE_CXX_STANDARD)
+set(CMAKE_CXX_STANDARD 17)
+# endif()
```
Comment (nim65s) [inline CMakeLists.txt]:
```suggestion
check_minimal_cxx_standard(17 ENFORCE)
```

Comment (olivier-stasse) [reply CMakeLists.txt]: Thanks done with commit [594cfe6](https://github.com/stack-of-tasks/dynamic_graph_bridge/pull/104/commits/594cfe67f8117c996c0deaebb2f5e27f78877f1b)

Comment (nim65s) [reply CMakeLists.txt]: for the record, you could also have clicked on the "commit suggestion" button here :)

**Pull request merged**